inc/budget.class.php

```diff
@@ -580,80 +683,142 @@ function showValuesByEntity() {
+                                       + `$costtable`.`cost_material`) AS value"
+               );
+               $criteria = [
+                  'SELECT'       => [
+                     $item->getTable() . '.entities_id',
+                     $sum
+                  ],
+                  'FROM'         => $costtable,
+                  'INNER JOIN'   => [
+                     $table => [
+                        'ON' => [
+                           $table      => 'id',
+                           $costtable  => $item->getForeignKeyField()
+                        ]
+                     ]
+                  ],
+                  'WHERE'        => [
+                     $costtable . '.budgets_id' => $budgets_id
+                  ] + getEntitiesRestrictCriteria($table, 'entities_id'),
+                  'GROUPBY'      => [
+                     $item->getTable() . '.entities_id'
+                  ]
+               ];
                break;
 
             default :
-               $query_infos = "SELECT SUM(`glpi_infocoms`.`value`) AS `sumvalue`,
-                                       `$table`.`entities_id`
-                                FROM `$table`
-                                INNER JOIN `glpi_infocoms`
-                                    ON (`glpi_infocoms`.`items_id` = `$table`.`id`
-                                        AND `glpi_infocoms`.`itemtype` = '$itemtype')
-                                WHERE `glpi_infocoms`.`budgets_id` = '$budgets_id' ".
-                                      getEntitiesRestrictRequest(" AND", $table, "entities_id");
-
+               $criteria = [
+                  'SELECT'       => [
+                     $table . '.entities_id',
+                     new QueryExpression('SUM(glpi_infocoms.value) AS sumvalue'),
+                  ],
+                  'FROM'         => $table,
+                  'INNER JOIN'   => [
+                     'glpi_infocoms' => [
+                        'ON' => [
+                           $table            => 'id',
+                           'glpi_infocoms'   => 'items_id'
+                        ]
+                     ]
+                  ],
+                  'WHERE'        => [
+                     'glpi_infocoms.itemtype'            => $itemtype,
+                     'glpi_infocoms.budgets_id'          => $budgets_id
+                  ] + getEntitiesRestrictCriteria($table, 'entities_id'),
+                  'GROUPBY'      => [
+                     $item->getTable() . '.entities_id'
+                  ]
+               ];
                if ($item->maybeTemplate()) {
-                  $query_infos .= " AND `$table`.`is_template` = 0 ";
+                  $criteria['WHERE'][$item->getTable() . '.is_template'] = 0;
+               }
+
+               if ($item instanceof Item_Devices) {
+                  $criteria['ORDERBY'][] = $item->getTable() .'.itemtype';
+               } else {
```

Comment:
order by name added ?

Reply:
There were no order in original code, will drop